### ISSUE TITLE
Download problem because of unicode

### ISSUE THREAD

**Issue by mgralle** [remar/adr_downloader#1]
Hi,

now that JMG has announced that the ADR will definitely disappear without a trace, I would very much like to use your script to download it all, especially including the comments! When running adr_downloader.py, I got the following bug on Mac Os 10.12.5 with Python 3.6.1:

Getting http://thearchdruidreport.blogspot.ca/2006/07/climbing-down-ladder.html
Traceback (most recent call last):
  File "\<stdin\>", line 1, in <module>
  File "adr_downloader-master/adr_downloader.py", line 20, in \<module\>
    f.write(post_data)
UnicodeEncodeError: 'ascii' codec can't encode character '\u2019' in position 1494: ordinal not in range(128)

Has this problem occurred on other OS? Actually, the script had been accessing lot of posts before that, but I can only find the first 5 in the raw folder.

Thanks for your attention and work!

**Comment by remar** [comment]
Hello!

No, I've never seen that issue before. I can have a look. I'm running
Lubuntu (Linux).

And it probably hasn't downloaded all those posts, I guess it's a bit
misleading. First it fetches all the references to all the posts (it has to
do a GET for every monthly archive), and then it goes through that list.

-Andreas
Den 26 maj 2017 16.32 skrev "mgralle" <notifications@github.com>:

**Comment by remar** [comment]
And I'm also thinking about adding functionality to parse out the comments
as well.
-Andreas
Den 26 maj 2017 17.20 skrev "Andreas Remar" <andreas.remar@gmail.com>:

**Comment by mgralle** [comment]
Hi,

for the Unicode error issue I found the following solution:

post_data = postdownloader.get_post(post_url)
post_data = post_data.replace(u"\u2018","'").replace(u"\u2019", "'")

Parsing the comments would be great! But having it all downloaded in one
place is the most important thing.

Cheers,

Matthias

2017-05-26 11:22 GMT-04:00 Andreas Remar <notifications@github.com>:

**Comment by remar** [comment]
Closing for now, please leave further feedback if issue isn't fixed.

**Comment by mgralle** [comment]
Hi,

actually the problem wasn't fixed by what I wrote above, I would have had to substitute a great number of Unicode characters manually. I think it is caused by the Python default console encoding, which is not utf-8 on my computer (French Canadian Mac). I managed to get everything working again by specifying utf-8 at every step and encoding-decoding Unicode explicitly. Looks a bit ugly but works now (files attached).

[utfsafe.zip](https://github.com/remar/adr_downloader/files/1036312/utfsafe.zip)

I would be very happy about parsing of the comments, but the important thing is that everything is now safe on my HD!

Thanks again,

Matthias
